include/OpenSpaceToolkit/Astrodynamics/Trajectory/TrajectorySegment.hpp

```diff
@@ -0,0 +1,161 @@
+{
+
+using ostk::core::ctnr::Array;
+using ostk::core::types::Shared;
+using ostk::core::types::String;
+
+using ostk::physics::time::Instant;
+using ostk::physics::time::Duration;
+
+using ostk::astro::trajectory::State;
+using ostk::astro::flight::system::Dynamics;
+using ostk::astro::EventCondition;
+
+/// @brief                                  Represent a propagation segment for astrodynamics purposes
+
+class TrajectorySegment
+{
+   public:
+    struct Solution
+    {
+       public:
+        Solution(const String& aName, const Array<Shared<Dynamics>>& aDynamicsArray, const Array<State>& aStates);
+
+        String name;                       /// Name of the segment.
+        Array<Shared<Dynamics>> dynamics;  /// List of dynamics used.
+        Array<State> states;               /// Array of states for the segment.
+    };
+
+    /// @brief                  Output stream operator
+    ///
+    /// @param                  [in] anOutputStream An output stream
+    /// @param                  [in] aSegment A TrajectorySegment
+    /// @return                 An output stream
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const TrajectorySegment& aSegment);
+
+    /// @brief                  Get name.
+    /// @return                 Name of the segment.
+
+    String getName() const;
+
+    /// @brief                  Get event condition.
+    /// @return                 Event condition.
+
+    Shared<EventCondition> getEventCondition() const;
+
+    /// @brief                  Get dynamics.
+    /// @return                 Dynamics.
+
+    Array<Shared<Dynamics>> getDynamics() const;
+
+    /// @brief                  Get numerical solver.
+    /// @return                 Numerical solver.
+
+    NumericalSolver getNumericalSolver() const;
+
+    /// @brief                  Access event condition.
+    /// @return                 Event condition.
+
+    const Shared<EventCondition>& accessEventCondition() const;
+
+    /// @brief                  Access dynamics.
+    /// @return                 Dynamics.
+
+    const Array<Shared<Dynamics>>& accessDynamics() const;
+
+    /// @brief                  Access numerical solver.
+    /// @return                 Numerical solver.
+
+    const NumericalSolver& accessNumericalSolver() const;
+
+    /// @brief                  Solve the segment
+    ///
+    /// @param                  [in] aState Initial state for the segment.
+    /// @param                  [in] aDynamicsArray Array of dynamics. Defaults to an empty array.
+    /// @param                  [in] aNumericalSolver Numerical solver to be used. Defaults to undefined.
+    /// @param                  [in] maximumPropagationDuration Maximum duration for propagation. Defaults to 30 days.
+    /// @return                 A Solution representing the result of the solve.
+
+    Solution solve(
+        const State& aState,
+        const Array<Shared<Dynamics>>& aDynamicsArray = Array<Shared<Dynamics>>::Empty(),
```

Comment:
Should this be called something like `extraDynamicsArray`? It looks like these get appended to whatever Dynamics the class is already holding.

Side-note: combining the dynamics and numerical solver (which are object bound things) with function input arguments feel a little weird to me. Seems like a strange mix of Object Oriented and Functional design. But maybe I'm missing something. What are your thoughts on reducing this to:
```
 Solution solve(
        const State& aState,
        const Duration& maximumPropagationDuration = Duration::Days(30.0)
    ) const;
```

And forcing dynamics/solver changes to use getters and setters?

Reply:
My biggest ick is that the numerical solver is an override here, while the dynamics are an addition, which is not very obvious IMO. 
But additionally the dynamics and integrator feel like core parts of the trajectory segment object, and so I think it makes sense to only treat them in an object-oriented way.

Let me know if something is going way over my head :)